src/NATS.Client.Core/NatsRequestSingleExtensions.cs

```diff
@@ -0,0 +1,83 @@
+using System.Buffers;
+using NATS.Client.Core.Internal;
+
+namespace NATS.Client.Core;
+
+public static class NatsRequestSingleExtensions
+{
+    // RequestAsync methods
+    // Same as PublishAsync with the following changes
+    // - Response is 0 (null) or 1 NatsMsg
+    // - PubOpts is called requestOpts
+    // - add SubOpts replyOpts
+    //   - default replyOpts.MaxMsgs to 1
+    //   - if replyOpts.Timeout == null then set to NatsOptions.RequestTimeout
+    public static async ValueTask<NatsMsg<TReply?>?> RequestSingleAsync<TRequest, TReply>(
```

Comment:
I think most of the other clients simply call it `Request` - not `RequestSingle` - [Reference](https://natsbyexample.com/examples/messaging/request-reply/go)